twill-yarn/src/test/java/org/apache/twill/yarn/EventHandlerTest.java

```diff
@@ -0,0 +1,351 @@
+      Assert.assertFalse(handlerFiles.contains(ABORTED_FILE));
+    } catch (Exception e) {
+      // kill the app as cleanup
+      controller.kill();
+    }
+  }
+
+  /**
+   * The handler for testing timeout handling.
+   */
+  public static final class Handler extends EventHandler {
+
+    private final String parentFolderPath;
+    private boolean abort;
+
+    public Handler(String parentFolderPath) {
+      this.parentFolderPath = parentFolderPath;
+    }
+
+    @Override
+    protected Map<String, String> getConfigs() {
+      return ImmutableMap.<String, String>builder()
+        .put("abort", "true")
+        .put("parentFolderPath", parentFolderPath)
+        .put("startedFile", STARTED_FILE)
+        .put("runFile", RUN_FILE)
+        .put("containerLaunched", CONTAINER_LAUNCHED_FOLDER)
+        .put("containerStopped", CONTAINER_STOPPED_FOLDER)
+        .put("completedFile", COMPLETED_FILE)
+        .put("killedFile", KILLED_FILE)
+        .build();
+    }
+
+    @Override
+    public void initialize(EventHandlerContext context) {
+      super.initialize(context);
+      this.abort = Boolean.parseBoolean(context.getSpecification().getConfigs().get("abort"));
+    }
+
+    @Override
+    public TimeoutAction launchTimeout(Iterable<TimeoutEvent> timeoutEvents) {
+      if (abort) {
+        return TimeoutAction.abort();
+      } else {
+        return TimeoutAction.recheck(10, TimeUnit.SECONDS);
+      }
+    }
+
+    @Override
+    public void started() {
+      try {
+        new File(context.getSpecification().getConfigs().get("parentFolderPath"),
+                 context.getSpecification().getConfigs().get("startedFile")).createNewFile();
+      } catch (IOException e) {
+        Throwables.propagate(e);
+      }
+    }
+
+    @Override
+    public void containerLaunched(String runnableName, int instanceId, String containerId) {
+      LOG.info("Launched {}#{} in container {}", runnableName, instanceId, containerId);
+      createContainerFile(runnableName, instanceId, containerId, "containerLaunched");
+    }
+
+    @Override
+    public void containerStopped(String runnableName, int instanceId, String containerId, int exitStatus) {
+      LOG.info("Stopped {}#{} in container {} with status {}", runnableName, instanceId, containerId, exitStatus);
+      createContainerFile(runnableName, instanceId, containerId, "containerStopped");
+    }
+
+    private void createContainerFile(String runnableName, int instanceId, String containerId, String folderKey) {
+      Map<String, String> configs = context.getSpecification().getConfigs();
+      File launchedFolder = new File(configs.get("parentFolderPath"), configs.get(folderKey));
+      if (!launchedFolder.exists()) {
+        launchedFolder.mkdirs();
+        launchedFolder.setReadable(true, false);
+      }
+      try {
+        new File(launchedFolder.getAbsolutePath(), Joiner.on(":").join(runnableName, instanceId, containerId))
```

Comment:
Would be clear if you have a private `touch(String name)` method rather than have `try-catch` block on each callback method.